config/initializers/health_check.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+HealthCheck.setup do |config|
+  # uri prefix (no leading slash)
+  config.uri = "health_check"
+
+  # Text output upon success
+  config.success = "success for following checks: ['database', 'migrations', 'email']"
+
+  # Timeout in seconds used when checking smtp server
+  config.smtp_timeout = 30.0
+
+  # http status code used when plain text error message is output
+  # Set to 200 if you want your want to distinguish between partial (text does not include success) and
+  # total failure of rails application (http status of 500 etc)
+
+  config.http_status_for_error_text = 500
+
+  # http status code used when an error object is output (json or xml)
+  # Set to 200 if you want your want to distinguish between partial (healthy property == false) and
+  # total failure of rails application (http status of 500 etc)
+
+  config.http_status_for_error_object = 500
+
+  # You can customize which checks happen on a standard health check, eg to set an explicit list use:
+  config.standard_checks = %w[database migrations email]
+
+  # Or to exclude one check:
+  # config.standard_checks -= [ 'emailconf' ]
+
+  # You can set what tests are run with the 'full' or 'all' parameter
+  config.full_checks = %w[database migrations email]
+
+  # Add one or more custom checks that return a blank string if ok, or an error message if there is an error
+  # config.add_custom_check do
+  #   CustomHealthCheck.perform_check # any code that returns blank on success and non blank string upon failure
+  # end
+
+  # Add another custom check with a name, so you can call just specific custom checks. This can also be run using
+  # the standard 'custom' check.
+  # You can define multiple tests under the same name - they will be run one after the other.
+  # config.add_custom_check('sometest') do
+  #   CustomHealthCheck.perform_another_check # any code that returns blank on success and non blank string upon failure
+  # end
+
+  # max-age of response in seconds
+  # cache-control is public when max_age > 1 and basic_auth_username is not set
+  # You can force private without authentication for longer max_age by
+  # setting basic_auth_username but not basic_auth_password
+  config.max_age = 1
+
+  # Protect health endpoints with basic auth
+  # These default to nil and the endpoint is not protected
+  # config.basic_auth_username = 'my_username'
+  # config.basic_auth_password = 'my_password'
```

Comment:
Do we care that this is on public internet? I guess we don't really, cause everything else is on public internet too. 

Do health-check requests show up in our logs? So that if someone tries a ddos through health-checks we will see it in our logs?

Reply:
Perhaps we could remove this comment as it reveals the password and username.

Reply:
Uhhh, I don't think this comment reveals much. My local username and password are different!

Reply:
This is from basic configuration and set as an example - personally - I'm not sure about passwords - from one side - yes it feels more secure - however Denial of Service attack could simply attack other pages - plus message stating everything is ok is of very little value. The question then becomes what should we reveal when things are not ok and are we happy with the outcome.

Reply:
> Denial of Service attack could simply attack other pages

I agree. Hence the question: do the healthcheck requests show up in our logs, like all other pages do (I think)?

Reply:
Yes the healthcheck does show up in our logs like any other request